### DISCUSSION TITLE
stdin via shell redirection?

### DISCUSSION THREAD

**Discussion by pkazmier** [zk-org/zk#239] [Help]
I was a bit surprised to find that I am unable to redirect standard input via shell redirection:

```sh
$ echo "Hello World" > /tmp/hello.txt
$ zk new --print-path --title Test < /tmp/hello.txt
/Users/kaz/notes/n4bz.md
$ cat /Users/kaz/notes/n4bz.md
# Test
```

Yet, I can pipe data into zk:

```sh
$ cat /tmp/hello.txt | zk new --print-path --title Test
/Users/kaz/notes/dvq7.md
$ cat /Users/kaz/notes/dvq7.md                         
# Test

Hello, World
```

I'm curious why zk goes [out of its way](https://github.com/mickael-menu/zk/blob/aaf6c42bd84f0319175d63785bccfe6360a71fe7/internal/util/os/os.go#L13) to only read from a pipe versus simply reading anything on stdin? It's a rare day when I find a command line tool that only accepts one form and not the other. I'm sure there is some valid reason, but I could not find any comments to explain this choice.

Thanks!

**Comment by mickael-menu** [comment]
@pkazmier For no good reason whatsoever! It's my first CLI tool (and Go project) and I somehow assumed that `os.Stdin` could not read piped input. Thank you for bringing this up.

Fixed in https://github.com/mickael-menu/zk/pull/240, would you mind checking it out?

**Comment by pkazmier** [comment]
@mickael-menu It works as I would expect new. Thank you!

For your first CLI tool, you hit the ball out of the park. You've converted me from my precious org-mode in emacs to switch to neovim, zk, zk-nvim, and taskwarrior.  I take a decent amount of notes as a manager (445 notes since 4/20/2020)—most are from bi-weekly one-on-ones with my team members. And while I have a highly customized org-mode workflow, tailored for those one-on-ones, I really like the zk interface (both command line and neovim integration) and the ability to link back and forth in my notes. Backlinks are a feature I've been missing as well as the ability to easily search my notes—zk fits the bill perfectly. And because you made it so extensible and well documented, over the past few days I've been able to tailor that workflow by adding command line completion of names and this morning I'm going to be writing my first neovim command `Zk1on1` that will open a telescope picker of anyone I've had prior one-on-ones with so I can quickly create a new meeting note for that individual. So, in short, you've killed my prior week because I'm having so much fun figuring out how to effectively use and customize this awesome tool. Thanks for all your hard work!

**Reply by mickael-menu** [reply]
Thank you for your kind words, I'm glad you're having fun using `zk`!

Feel free to share your custom Neovim commands on a new discussion in `zk-nvim`, that could be interesting to others as well.
